Packs/ProofpointEmailSecurity/Integrations/ProofpointEmailSecurityEventCollector/ProofpointEmailSecurityEventCollector.py

```diff
@@ -0,0 +1,141 @@
+    """
+    This function fetches events from the websocket connection for the given event type, for the given fetch interval
+
+    Args:
+        event_type (EventType): The event type to fetch (MAILLOG, MESSAGE)
+        connection (Connection): the websocket connection to the event type
+        fetch_interval (int): the interval of events to fetch, in seconds
+
+    Returns:
+        list[dict]: A list of events
+    """
+    events: list[dict] = []
+    event_ids = set()
+    fetch_start_time = datetime.utcnow()
+    while not is_interval_passed(fetch_start_time, fetch_interval):
+        try:
+            # wait for events for the remaining time of the interval
+            timeout = fetch_interval - (datetime.utcnow() - fetch_start_time).total_seconds() + 1
+            event = json.loads(connection.recv(timeout=timeout))
+        except TimeoutError:
+            # if we didn't receive an event for `fetch_interval` seconds, finish fetching
+            break
+        event_id = event.get("id", event.get("guid"))
+        event_ts = event.get("ts")
+        if not event_ts:
+            # if timestamp is not in the response, use the current time
+            demisto.debug(f"Event {event_id} does not have a timestamp, using current time")
+            event_ts = datetime.utcnow().isoformat()
+        date = dateparser.parse(event_ts)
+        if not date:
+            demisto.debug(f"Event {event_id} has an invalid timestamp, using current time")
+            # if timestamp is not in correct format, use the current time
+            date = datetime.utcnow()
+        # the `ts` parameter is not always in UTC, so we need to convert it
+        event["_time"] = date.astimezone(tz.tzutc()).isoformat()
+        event["event_type"] = event_type.value
+        events.append(event)
+        event_ids.add(event_id)
+    demisto.debug(f"Fetched {len(events)} events of type {event_type}")
+    demisto.debug("The fetched events ids are: " + ", ".join([str(event_id) for event_id in event_ids]))
+    return events
+
+
+def test_module(host: str, cluster_id: str, api_key: str):
+    # set the fetch interval to 10 seconds so we don't get timeout for the test module
+    fetch_interval = 10
+    with websocket_connections(host, cluster_id, api_key) as (message_connection, maillog_connection):
+        fetch_events(EventType.MESSAGE, message_connection, fetch_interval)
+        fetch_events(EventType.MAILLOG, maillog_connection, fetch_interval)
+        return "ok"
+
+
+def long_running_execution_command(host: str, cluster_id: str, api_key: str, fetch_interval: int):
+    with websocket_connections(host, cluster_id, api_key) as (message_connection, maillog_connection):
+        demisto.info("Connected to websocket")
+        message_events = fetch_events(EventType.MESSAGE, message_connection, fetch_interval)
+        maillog_events = fetch_events(EventType.MAILLOG, maillog_connection, fetch_interval)
```

Comment:
if I understand correctly, since this is serial, we will fetch messages for 1 minute, then stop, then fetch mail logs for 1 minute then ingest, and start over. that means we have an effective interval of 2 minutes.
let's talk about this.

Reply:
yes, that's the current implemntation.
Our options:
1. keep this implementation.
2. fetch_events should be given `fetch_interval / 2`.
3. Map the `fetch_events` function into two threads.